utils/devel/sync-fork.sh

```diff
@@ -42,14 +44,15 @@ if ! [[ $AvailableRemotes =~ $Upstream ]]; then
     git remote add upstream "$Upstream"
 fi
 
-git fetch upstream || Error "Fetching Upstream."
-
+CurrentBranch=`git rev-parse --abbrev-ref HEAD`
 
 git checkout master || Error "Checking out Master."
```

Comment:
@sbts I now see that you're checking out master here first, which is good. However, you can't always check out master: an error will be thrown when the current working directory isn't clean and conflicts arise. Better stash before checkout and then ```stash pop``` after returning to the branch.